cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -630,18 +631,17 @@ private List<ApplicationId> decodeAndValidateBatchApplication(NamespaceId namesp
           throw new BadRequestException("Missing 'appId' in the request element.");
         }
         String appId = obj.get("appId").getAsString();
-        String version;
         JsonElement versionObj = obj.get("version");
-        if (versionObj == null || versionObj.equals("-SNAPSHOT")) {
+        if (versionObj == null || versionObj.getAsString().equals("-SNAPSHOT")) {
           // get the latest version of the app instead of -SNAPSHOT
-            ApplicationDetail appLatestVersionDetail = applicationLifecycleService.getLatestAppDetail
-              (new ApplicationId(namespaceId.getNamespace(), appId));
-            version = appLatestVersionDetail.getAppVersion();
+          appNames.add(appId);
```

Comment:
@de-lan this changed the order of returned response, causing the test failure. Do we care about the order?

Reply:
The result is in a Jason array, the order perhaps don't matter